src/librustc/mir/interpret/mod.rs

```diff
@@ -554,3 +556,70 @@ pub fn truncate(value: u128, size: Size) -> u128 {
     // Truncate (shift left to drop out leftover values, shift right to fill with zeroes).
     (value << shift) >> shift
 }
+
+
+impl<'tcx> TyCtxt<'tcx> {
+
+    /// Evaluates a constant without providing any substitutions. This is useful to evaluate consts
+    /// that can't take any generic arguments like statics, const items or enum discriminants.
+    pub fn const_eval_mono(self, def_id: DefId) -> ConstEvalResult<'tcx> {
+        let substs = InternalSubsts::identity_for_item(self, def_id);
+        let instance = ty::Instance::new(def_id, substs);
```

Comment:
Isn't there a `ty::Instance::mono` or similar?

Reply:
In some situations like enum discriminants and repeat counts, generic parameters can be within scope but they aren't allowed to be used within the constant. `ty::Instance::mono` asserts that `def_id` has no non-lifetime substs, but if identity substitutions are created the constant can be successfully evaluated unless `const_eval` encounters one of the generic parameters.

Reply:
Okay, in that case I wouldn't call it "mono", but rather, perhaps, "poly" - although ideally this method wouldn't be needed.